Common/src/geometry_structure.cpp

```diff
@@ -15564,49 +15564,349 @@ void CPhysicalGeometry::SetColorGrid_Parallel(CConfig *config) {
-      Point_2 = elem[iElem]->GetNode(1); Coord_2 = node[Point_2]->GetCoord();
-      Point_3 = elem[iElem]->GetNode(2); Coord_3 = node[Point_3]->GetCoord();
+    /*--- Point identification, edge normal vector and area ---*/
+    
+    const unsigned long iPoint = edge[iEdge]->GetNode(0);
+    const unsigned long jPoint = edge[iEdge]->GetNode(1);
+    
+    const unsigned long GlobalIndex_i = node[iPoint]->GetGlobalIndex();
+    const unsigned long GlobalIndex_j = node[iPoint]->GetGlobalIndex();
+    
+    const su2double *Normal = edge[iEdge]->GetNormal();
+    
+    /*--- Face area (norm of the normal vector) ---*/
+    
+    su2double Area = 0.0;
+    for (unsigned short iDim = 0; iDim < nDim; iDim++)
+      Area += Normal[iDim]*Normal[iDim];
+    Area = sqrt(Area);
+    
+    /*--- Aspect ratio is the ratio between the largest and smallest
+     faces making up the boundary of the dual CV and is a
+     measure of the aspect ratio of the dual control volume. Smaller
+     is better (closer to isotropic). ----*/
+    
+    if (node[iPoint]->GetDomain()) {
+      if (Area < Area_Min[iPoint]) Area_Min[iPoint] = Area;
+      if (Area > Area_Max[iPoint]) Area_Max[iPoint] = Area;
+    }
+    
+    if (node[jPoint]->GetDomain()) {
+      if (Area < Area_Min[jPoint]) Area_Min[jPoint] = Area;
+      if (Area > Area_Max[jPoint]) Area_Max[jPoint] = Area;
+    }
+    
+    if (Area == 0.0) {
+      char buf[200];
+      SPRINTF(buf, "Zero-area CV face found for edge (%lu,%lu).",
+              GlobalIndex_i, GlobalIndex_j);
+      SU2_MPI::Error(string(buf), CURRENT_FUNCTION);
+    }
+    
+    /*-- Unit normal for the current edge. Recall that this normal
+     is computed by summing the normals or adjacent faces along
+     the edge between iPoint & jPoint. ---*/
+    
+    vector<su2double> UnitNormal(nDim);
+    for (unsigned short iDim = 0; iDim < nDim; iDim++)
+      UnitNormal[iDim] = Normal[iDim]/Area;
+    
+    /*--- Get the coordinates for point i & j. ---*/
+    
+    const su2double *Coord_i = node[iPoint]->GetCoord();
+    const su2double *Coord_j = node[jPoint]->GetCoord();
+    
+    /*--- Compute vector pointing from iPoint to jPoint and the distance. ---*/
+    
+    su2double dist_ij = 0.0;
+    vector<su2double> Edge_Vector(nDim);
+    for (unsigned short iDim = 0; iDim < nDim; iDim++) {
+      Edge_Vector[iDim] = Coord_j[iDim]-Coord_i[iDim];
+      dist_ij += Edge_Vector[iDim]*Edge_Vector[iDim];
+    }
+    dist_ij = sqrt(dist_ij);
+    
+    /*-- Normalize the edge vector. ---*/
+    
+    vector<su2double> Unit_Edge_Vector(nDim);
+    for (unsigned short iDim = 0; iDim < nDim; iDim++)
+      Unit_Edge_Vector[iDim] = Edge_Vector[iDim]/dist_ij;
+    
+    /*--- Compute the angle between the unit normal associated
+     with the edge and the unit vector pointing from iPoint to jPoint. ---*/
+    
+    su2double dot_prod = 0.0;
+    for (unsigned short iDim = 0; iDim < nDim; iDim++)
```

Comment:
Some of these loops for areas, edge vectors and dot prod can be fused.

Reply:
Rearranged some of the smaller loops for efficiency